File: python/src/dependency/interface.py
# internal
from dependency.mathcalculation import MathCalculation


class Interface(object):

    def __init__(self):
        pass

    def compute_service(
        self,
        service_name: str,
        service: MathCalculation,
        num: int
    ):
        if (not service_name
                or not isinstance(service_name, str)
                or not isinstance(service, MathCalculation)
                or not num
                or not isinstance(num,int)):
            return
        
        result: int = service.compute(num)

        print(service_name + " of " + str(num) + ": " + str(result))

File: java/src/main/java/dependency/injection/java/dependency/Interface.java
package dependency.injection.java.dependency;

public class Interface {

    public Interface() {}

    public void computeService(
        String serviceName,
        MathCalculation service,
        int num
    ) {
        long result = service.compute(num);
        System.out.println(serviceName + " of "
                            + String.valueOf(num)
                            + ": " + result);
    }
    
}
